src/OrbitGl/App.cpp

```diff
@@ -1684,18 +1687,22 @@ orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModu
         std::chrono::steady_clock::now();
     ORBIT_LOG("Copying \"%s\" started", debug_file_path);
     orbit_base::Future<ErrorMessageOr<void>> copy_result =
-        secure_copy_callback_(debug_file_path, local_debug_file_path.string());
+        secure_copy_callback_(debug_file_path, local_debug_file_path.string(), stop_token);
 
     orbit_base::ImmediateExecutor immediate_executor{};
     return copy_result.Then(
         &immediate_executor,
         [debug_file_path, local_debug_file_path, scoped_status = std::move(scoped_status),
-         copy_begin](ErrorMessageOr<void> sftp_result) -> ErrorMessageOr<std::filesystem::path> {
+         copy_begin, stop_token = std::move(stop_token)](
+            ErrorMessageOr<void> sftp_result) -> ErrorMessageOr<std::filesystem::path> {
           if (sftp_result.has_error()) {
             return ErrorMessage{
                 absl::StrFormat("Could not copy debug info file from the remote: %s",
                                 sftp_result.error().message())};
           }
+          if (stop_token.IsStopRequested()) {
+            return ErrorMessage{"User canceled download."};
+          }
```

Comment:
Oh I see you're handling the cancel case here. I think this should be moved into ServiceDeployManager where `promise.SetResult` gets called.

Reply:
done